src/main/java/org/generationcp/middleware/dao/LocationDAO.java

```diff
@@ -992,85 +993,107 @@ public long countByProgramUUID(final String programUUID) {
+			sqlQuery.setParameter("locationId", locationSearchRequest.getLocationId());
+		}
+	}
 
-			if (!results.isEmpty()) {
-				for (final Object[] row : results) {
-					final Integer locationDbId = (Integer) row[0];
-					final String locationType = (String) row[1];
-					final String name = (String) row[2];
-					final String abbreviation = (String) row[3];
-					final String countryCode = (String) row[4];
-					final String countryName = (String) row[5];
-					final Double latitude = (Double) row[6];
-					final Double longitude = (Double) row[7];
-					final Double altitude = (Double) row[8];
-
-					final LocationDetailsDto locationDetailsDto =
-						new LocationDetailsDto(locationDbId, locationType, name, abbreviation, countryCode, countryName, latitude,
-							longitude, altitude);
-					locationDetailsDto.setLocationName(name);
-					if (!locationType.equalsIgnoreCase(LocationDAO.COUNTRY)) {
-						final AdditionalInfoDto additionalInfoDto = new AdditionalInfoDto(locationDetailsDto.getLocationDbId());
-						additionalInfoDto.addInfo("province", (String) row[9]);
-						locationDetailsDto.setMapAdditionalInfo(additionalInfoDto);
-					}
-					locationList.add(locationDetailsDto);
-				}
-			}
+	private String createGetLocationsQuery(final LocationSearchRequest locationSearchRequest) {
+		final StringBuilder queryString = new StringBuilder();
+		queryString.append("SELECT l.locid AS locationDbId, ");
+		queryString.append("ud.fname AS locationType, ");
+		queryString.append("l.lname AS locationName, ");
+		queryString.append("l.labbr AS abbreviation, ");
+		queryString.append("c.isothree AS countryCode, ");
+		queryString.append("c.isoabbr AS countryName, ");
+		queryString.append("g.lat AS latitude, ");
+		queryString.append("g.lon AS longitude, ");
+		queryString.append("g.alt AS altitude, ");
+		queryString.append("province.lname AS province ");
+		this.appendGetLocationFromQuery(queryString);
+		this.appendLocationSearchFilter(queryString, locationSearchRequest);
+		queryString.append(" ORDER BY l.locid ");
+		return queryString.toString();
+	}
 
-			return locationList;
+	private void appendGetLocationFromQuery(final StringBuilder queryString) {
+		queryString.append("FROM location l ");
+		queryString.append(" LEFT JOIN georef g on l.locid = g.locid ");
+		queryString.append(" LEFT JOIN cntry c on l.cntryid = c.cntryid ");
+		queryString.append(" LEFT JOIN udflds ud on ud.fldno = l.ltype ");
+		queryString.append(" ,location province ");
+	}
 
-		} catch (final HibernateException e) {
-			LocationDAO.LOG.error(e.getMessage(), e);
-			throw new MiddlewareQueryException(
-				this.getLogExceptionMessage("getLocalLocationsByFilter", "", null, e.getMessage(), LocationDAO.CLASS_NAME_LOCATION), e);
+	private void appendLocationSearchFilter(final StringBuilder queryString, final LocationSearchRequest locationSearchRequest) {
+		queryString.append("WHERE province.locid = l.snl1id ");
```

Comment:
You can `WHERE 1 = 1` to initialaze a  WHERE clause, regardless if search request had a filter